cel/library.go

```diff
@@ -842,11 +830,8 @@ func (i *ipv6) zoneID() bool {
 func (i *ipv6) dotted() bool {
 	start := i.index
 	i.dottedRaw = ""
-	for {
-		if i.index < len(i.str) && (i.digit() || i.take('.')) {
-			continue
-		}
-		break
+	for i.digit() || i.take('.') {
+		// Consume '*( DIGIT "." )'
```

Comment:
This is one place where we're a bit loose w/ the parsing - basically parsing any 7+ length sequence of digits / '.' into a potential IP address. In other places in the parser we're a bit more strict where we don't advance the parser's position if we don't match the expected sequences.

I don't think this is an issue in practice, but potentially a place where we could make it consistent.

Reply:
Potentially, IPv6 and IPv4 parsers could be merged, which would also remove the extra allocation https://github.com/bufbuild/protovalidate-go/pull/202#issuecomment-2785946148.

Reply:
Merging the IPv4 and IPv6 parsers make sense to me; in protovalidate-cc where I didn't follow the TS implementation exactly I did share some of the logic between them anyway.